discovery/balancer.go

```diff
@@ -126,6 +128,7 @@ func (b *watcherBalancer) WaitForTransition(ctx context.Context, to Addr) error
 
 // hasTransitioned checks if we've finished transitioning to the given address.
 func (b *watcherBalancer) hasTransitioned(to Addr) error {
+	// FIXME: why does this panic?
```

Comment:
This seems to be because `watcher.balancer` is no longer set in `balancerBuilder.Build`, and so is nil when attempting to call `w.balancer.WaitForTransition` in `watcher.switchServer`